Hugging Face/Hugging_Face_Few_Shot_Learning_with_Inference_API.ipynb

```diff
@@ -0,0 +1,643 @@
+     "iopub.status.busy": "2023-11-08T19:11:34.006521Z",
+     "iopub.status.idle": "2023-11-08T19:11:34.010795Z",
+     "shell.execute_reply": "2023-11-08T19:11:34.010166Z",
+     "shell.execute_reply.started": "2023-11-08T19:11:34.006746Z"
+    },
+    "papermill": {},
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "MODEL = \"EleutherAI/gpt-neo-1.3B\"\n",
+    "API_TOKEN = \"<INSERT_API_TOKEN>\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ec280229",
+   "metadata": {
+    "papermill": {},
+    "tags": []
+   },
+   "source": [
+    "## Model"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "38e868fc",
+   "metadata": {
+    "papermill": {},
+    "tags": []
+   },
+   "source": [
+    "### Define function to make API calls to Hugging Face endpoints"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "37c02550",
+   "metadata": {
+    "execution": {
+     "iopub.execute_input": "2023-11-08T18:47:21.815058Z",
+     "iopub.status.busy": "2023-11-08T18:47:21.814813Z",
+     "iopub.status.idle": "2023-11-08T18:47:21.821996Z",
+     "shell.execute_reply": "2023-11-08T18:47:21.821243Z",
+     "shell.execute_reply.started": "2023-11-08T18:47:21.815034Z"
+    },
+    "papermill": {},
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "def query(\n",
+    "        payload='', \n",
+    "        model = 'EleutherAI/gpt-neo-1.3B', \n",
+    "        parameters = {\n",
+    "            'max_new_tokens':5,\n",
+    "            'temperature': 0.5\n",
+    "        }, \n",
+    "        options = {\n",
+    "            'use_cache': False\n",
+    "        }\n",
+    "    ):\n",
+    "    API_URL = f\"https://api-inference.huggingface.co/models/{model}\"\n",
+    "    headers = {\"Authorization\": f\"Bearer {API_TOKEN}\"}\n",
+    "    body = {\"inputs\":payload,'parameters':parameters,'options':options}\n",
+    "    \n",
+    "    try:\n",
+    "        response = requests.request(\"POST\", API_URL, headers=headers, data= json.dumps(body))\n",
+    "        return response.json()[0]['generated_text']\n",
+    "    except:\n",
+    "        return \"Error: \" + \" \".join(response.json()['error'])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "378de1e9",
+   "metadata": {
+    "execution": {
+     "iopub.execute_input": "2022-11-02T22:04:21.973105Z",
+     "iopub.status.busy": "2022-11-02T22:04:21.972816Z",
+     "iopub.status.idle": "2022-11-02T22:04:21.980111Z",
+     "shell.execute_reply": "2022-11-02T22:04:21.977454Z",
+     "shell.execute_reply.started": "2022-11-02T22:04:21.973076Z"
+    },
+    "papermill": {},
+    "tags": []
+   },
+   "source": [
+    "## Output"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f0724801-389c-4184-b3a1-a3491573e24e",
+   "metadata": {},
+   "source": [
+    " The model usually takes time to load in the hugging face server. For example, model gpt-neo-1.3B takes approximately 212 seconds"
```

Comment:
Make sure to bold this as the remaining part of the notebook depends on this.

Reply:
Right, I will bold it.